sycl/include/CL/sycl/detail/image_accessor_util.hpp

```diff
@@ -464,97 +448,74 @@ void convertReadData(vec<ChannelType, 4> PixelData,
 // int32, uint32, half. ImageChannelType is passed to the function to use
 // appropriate conversion rules.
 template <typename ChannelType>
-vec<ChannelType, 4> convertWriteData(vec<cl_uint, 4> WriteData,
-                                     image_channel_type ImageChannelType) {
-  vec<ChannelType, 4> PixelData;
-  vec<cl_uint, 4> Temp;
-  cl_uint MinVal;
-  cl_uint MaxVal;
+vec<ChannelType, 4>
+convertWriteData(const vec<cl_uint, 4> WriteData,
+                 const image_channel_type ImageChannelType) {
   switch (ImageChannelType) {
-  case image_channel_type::unsigned_int8:
+  case image_channel_type::unsigned_int8: {
     // convert_uchar_sat(Data)
-    MinVal = min_v<cl_uchar>();
-    MaxVal = max_v<cl_uchar>();
-    Temp = (cl::sycl::clamp(WriteData, MinVal, MaxVal));
-    PixelData = Temp.convert<ChannelType>();
-    break;
-  case image_channel_type::unsigned_int16:
+    cl_uint MinVal = min_v<cl_uchar>();
+    cl_uint MaxVal = max_v<cl_uchar>();
+    vec<cl_uint, 4> PixelData = cl::sycl::clamp(WriteData, MinVal, MaxVal);
+    return PixelData.convert<ChannelType>();
+  }
+  case image_channel_type::unsigned_int16: {
     // convert_ushort_sat(Data)
-    MinVal = min_v<cl_ushort>();
-    MaxVal = max_v<cl_ushort>();
-    Temp = cl::sycl::clamp(WriteData, MinVal, MaxVal);
-    PixelData = Temp.convert<ChannelType>();
-    break;
+    cl_uint MinVal = min_v<cl_ushort>();
+    cl_uint MaxVal = max_v<cl_ushort>();
+    vec<cl_uint, 4> PixelData = cl::sycl::clamp(WriteData, MinVal, MaxVal);
+    return PixelData.convert<ChannelType>();
+  }
   case image_channel_type::unsigned_int32:
     // no conversion is performed.
-    PixelData = WriteData.convert<ChannelType>();
-    break;
+    return WriteData.convert<ChannelType>();
   default:
     // OpenCL Spec section 6.12.14.4 does not allow writing uint4 data to an
     // image with channel datatype other than unsigned_int8,unsigned_int16 and
     // unsigned_int32.
     throw cl::sycl::invalid_parameter_error(
         "Datatype of data to write - cl_uint4 is incompatible with the "
         "image_channel_type of the image.");
-    break;
   }
-  return PixelData;
 }
 
 template <typename ChannelType>
-vec<ChannelType, 4> convertWriteData(vec<cl_int, 4> WriteData,
-                                     image_channel_type ImageChannelType) {
+vec<ChannelType, 4>
+convertWriteData(const vec<cl_int, 4> WriteData,
+                 const image_channel_type ImageChannelType) {
 
-  vec<ChannelType, 4> PixelData;
-  vec<cl_int, 4> Temp;
+  vec<cl_int, 4> PixelData;
```

Comment:
For `convertWriteData(const vec<cl_uint, 4>` you applied my comments, but not here.

Reply:
Because it was used in two of the three cases.
I'll do the change.

Reply:
I realised it was also the case for the convertWriteData with cl_uint argument. I may have missed it.